include/blas_meta.h

```diff
@@ -136,6 +135,30 @@ static SYCL_BLAS_INLINE index_t roundUp(index_t x, index_t y) {
   return ((x + y - 1) / y) * y;
 }
 
+template <typename index_t, typename vector_t>
+index_t vec_total_size(index_t &vector_size, vector_t &&current_vector) {
+  vector_size += static_cast<index_t>(current_vector.size());
+  return 0;
+}
+
+template <typename vector_t>
+int append_vector(vector_t &lhs_vector, vector_t const &rhs_vector) {
+  lhs_vector.insert(lhs_vector.end(), rhs_vector.begin(), rhs_vector.end());
+  return 0;
+}
+
+template <typename first_vector_t, typename... other_vector_t>
+first_vector_t concatenate_vectors(first_vector_t first_vector,
+                                   other_vector_t &&... other_vectors) {
+  int first_Vector_size = static_cast<int>(first_vector.size());
+  int s[] = {vec_total_size(first_Vector_size, other_vectors)..., 0};
```

Comment:
I realise this is not really relevant to this PR, but why do `vec_total_size`/`append_vector` return 0 and get assigned to arrays that are never used?